vald/evm/evm.go

```diff
@@ -75,24 +97,21 @@ func (mgr Mgr) ProcessNewChain(event *types.ChainAdded) (err error) {
 
 // ProcessDepositConfirmation votes on the correctness of an EVM chain token deposit
 func (mgr Mgr) ProcessDepositConfirmation(event *types.ConfirmDepositStarted) error {
-	if !mgr.isParticipantOf(event.Participants) {
-		mgr.logger("pollID", event.PollID).Debug("ignoring deposit confirmation poll: not a participant")
-		return nil
-	}
-
-	txReceipt, err := mgr.GetTxReceiptIfFinalized(event.Chain, common.Hash(event.TxID), event.ConfirmationHeight)
-	if err != nil {
-		return err
-	}
-	if txReceipt == nil {
-		mgr.logger().Infof("broadcasting empty vote for poll %s", event.PollID.String())
-		_, err := mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain)))
-
-		return err
-	}
+	vc := voteContext{
+		Participants:       event.Participants,
+		PollMappings:       []types.PollMapping{{PollID: event.PollID, TxID: event.TxID}},
+		Chain:              event.Chain,
+		ConfirmationHeight: event.ConfirmationHeight,
+		PollType:           "token deposit",
+	}
+	return mgr.vote(vc, func(logs []*geth.Log) []types.Event {
+		return mgr.processTokenDepositLogs(logs, event)
```

Comment:
annoying that go doesn't support generic in struct methods

Reply:
yes!